### ISSUE TITLE
@Endcond not behaving as expected (missing warnings, missing output) 

### ISSUE THREAD

**Issue by lcmrvk** [doxygen/doxygen#8393]
**Describe the bug**

I'm including both of these issues here because I think they might be related.

1. One is that the missing endcondition warning does is not reported in every case in which I'd
expected to. It does get reported in some cases however.

See:
- example1
- example2
- example3

2. There was a bug https://github.com/doxygen/doxygen/issues/3857 that was fixed in 1.7.2 where the fix automatically inserted an `@endcond` so that it does not affect content in files which include the file with the missing `@endcond`.

See:
- example4

See below examples with their expected and actual behavior. 

**Expected behavior**

**example1:**

todo:
- remove `@endcond` in `A.h` which belongs to `#define w`

expected:
- missing endcondition warning

result:
-  no missing `@endcond` warning, however `@endcond` seems to be added automatically to the end of the file because `B.h` (which includes `A.h`) is not affected
- output is missing everything from `A.h` as expected

 ---

**example2:**

todo:

- remove `@endcond` in `B.h` which belongs to `#define k`

expected:
-  missing endcondition warning

result:
- warning is reported as expeced
- `#define l` missing from output as expected

  ---

**example3:**

todo:
- remove `@endcond` in `A.h` which belongs to `#define y`

expected:
-  missing endcondition warning

result:   
- no missing `@endcond` warning, however `@endcond` seems to be added automatically to the end of the file because `B.h` (which includes `A.h`) is not affected
- `#define x` (`from A.h`) is correctly included in output

  ---

**example4:**

todo:
-  remove `@endcond` in `A.h` which belongs to` #define z`

expected:
- missing endcondition warning

result:
- missing `@endcond` warning reported
- `#define j` (from `B.h`), which should be included in the output is missing
- `#define l` is included in the output as expected

**To Reproduce**

The steps to reproduce the examples are given above.
The commands to be executed are as follows:
1. `cd <folder-with-atached-files>`
2. `doxygen`
3. `cd latex`
4. `make`

Additional packages that need to be installed are:
- `make`
- `texlive-base`
- `texlive-latex-extra`

**Version**
doxygen version: `1.9.1`
linux version: (tested in docker)
```
Distributor ID:	Debian
Description:	Debian GNU/Linux bullseye/sid
Release:	testing
Codename:	bullseye
```

[doxygen-endcond-test.tar.gz](https://github.com/doxygen/doxygen/files/6009070/doxygen-endcond-test.tar.gz)

**Comment by albert-github** [comment]
In the current doxygen version (1.9.7) we get warnings about the missing `@endcond` for the examples 1, 3 and 4.
The example 2 is and was no problem (as also written in the issue).

We see though that with the example2 1, 3 and 4 we also get an incorrect warning  like:
```
B.h:5: warning: Conditional section with label 'INCLUDE_INTERNAL_CODE' does not have a corresponding \endcond command within this file.
```
When a warning should be given it should not be `B.h` but `A.h`

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10156

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.8.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
